### PR TITLE
Porting TPC Data compression decoding to GPU

### PR DESCRIPTION
**Pull request by cima22**
Average of 300 clusters over 53175874 not correctly decoded, maybe fast-math problem. Still need to incorporate GPU decoding in the offline chain.

### CODE REVIEW

**Review by davidrohr — DISMISSED** [review]
Approving to start the CI, please do not merge yet!
The code-formatting check fails. Could you please run git-clang-format, to fix the formatting. Please also squash all test commits, and leave only meaningfull commits in. Alternatively, we can squash the full PR when merging.

File: CMakeLists.txt
Comment (davidrohr) [inline CMakeLists.txt]: should not be part of the PR, please keep as local test commit

File: GPU/CMakeLists.txt
Comment (davidrohr) [inline GPU/CMakeLists.txt]: should not be part of the PR, please keep as local test commit

File: GPU/GPUTracking/CMakeLists.txt
Comment (davidrohr) [inline GPU/GPUTracking/CMakeLists.txt]: should not be part of the PR, please keep as local test commit

File: GPU/GPUTracking/DataCompression/GPUTPCDecompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/DataCompression/GPUTPCDecompression.cxx]: The `SetPointersTmpNativeBuffersGPU` is only needed temporarily while your decoding is running? Then you should add the flag MEMORY_SCRATCH and remove MEMORY_GPU. The scratch memory will automatically be allocated on the GPU if the processing step runs on the GPU and on the host otherwise. This should also allow your code to run as CPU version transparently.

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: I think a newline at the end of the file is missing

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: Aren't you leaking the original ptr? Perhaps better use a unique_ptr?

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: No need to store copySize in a variable

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: You do not need to synchrinize with this stream I believe? Nothing is happening in this stream before.

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: The name outputStream is a bit misleading, since you are using it for the input?

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: You run the kernel in the stream 0. You should make sure this is the same stream as above, where you have the stream id defined as outputStream. I'd use the variable also here, otherwise it gets confusing.

File: GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingCompression.cxx]: This seems to be only test code, or am I wrong? If this is needed, perhaps move it to `GPUChainTrackingDebugAndProfiling.cxx`?

File: GPU/GPUTracking/Standalone/Benchmark/standalone.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Standalone/Benchmark/standalone.cxx]: This should not be part of the PR I believe?

**Review by davidrohr — COMMENTED** [review]
Looks pretty good so far, though I didn't fully check the functionality. I have mostly a couple of cosmetic comments.

**Review by davidrohr — APPROVED** [review]
approved to start the CI

File: GPU/GPUTracking/Standalone/Benchmark/standalone.cxx
Comment (davidrohr) [inline GPU/GPUTracking/Standalone/Benchmark/standalone.cxx]: please don't reformat files you didn't touch in the PR!

File: GPU/GPUTracking/Base/cuda/CMakeLists.txt
Comment (davidrohr) [inline GPU/GPUTracking/Base/cuda/CMakeLists.txt]: please merge this with the cmake command above

**Review by davidrohr — DISMISSED** [review]
Looks good, let's see what the CI says. I have only 2 cosmetic comments below, which can also be fixed afterwards.

**Comment by davidrohr** [comment]
@ktf : It seems framework core ctest is failing regularly in the CI today, this seems to be a new problem?

**Comment by ktf** [comment]
I am checking.

**Comment by ktf** [comment]
Should be fixed by #12781 .

File: GPU/GPUTracking/Definitions/GPUDefGPUParameters.h
Comment (davidrohr) [inline GPU/GPUTracking/Definitions/GPUDefGPUParameters.h]: For reference: Did you actually tune these values, also the ones below, for the respective GPU models?
Or did you just copy them?
If they are not tuned, we should remove them here, and just leave 
```
#ifndef GPUCA_LB_GPUTPCDecompressionKernels_step0attached
    #define GPUCA_LB_GPUTPCDecompressionKernels_step0attached 256
  #endif
```
etc below.

Comment (cima22) [reply GPU/GPUTracking/Definitions/GPUDefGPUParameters.h]: I did not tune them, I just copied the values from the compression kernels. I will remove them and leave just the general ones below.

**Review by davidrohr — APPROVED** [review]
can be merged now if the FullCI becomes green

**Comment by davidrohr** [comment]
Full CI again failed randomly. Tried it locally and it worked, and also works as part of #12799, so should be fine, merging.

**Pull request merged**
